### PR TITLE
luci-app-upnp: Finalise changes for OpenWrt 24.10 release

### PR DESCRIPTION
**Pull request by Self-Hosting-Group**
- Complete plugin wording/UI revision
  Add missing protocol term in status/overview UI and shorten menu option to `UPnP IGD & PCP`, otherwise two lines, rearrange table columns, improve wording and add help texts and note upstream change miniupnp/miniupnp@9339f0e
- Simplify UI by moving rarely used options to advanced settings, rest to service setup and rearrangement
- Hide dependent UI fields if UPnP IGD protocol is disabled
- Remove `clean_ruleset_interval/threshold` UI options as not working
- Require client address and improve defaults in ACL entry UI
- Manual adaption of changed non-translatable protocol acronyms in translations, see #7175

This PR completes committed changes #6863 (merged in #6975) and continues #7217 for the UPnP IGD & PCP/NAT-PMP autonomous port mapping service.

### Service setup UI:

openwrt-settings
_[Screenshot 23.05 release](https://github.com/user-attachments/assets/38cbb287-e85d-4fd8-bed6-47a3a4a2cfbc)_

### Advanced settings UI:

openwrt-advanced-settings
_[Screenshot 23.05 release](https://github.com/user-attachments/assets/d0b9bcfc-d3af-4941-af6a-19e5bfcd3f81)_

### ACL entry UI:

openwrt-acl-entry

### Status/overview UI:

openwrt-status-overview

The **Port Control Protocol (PCP)** is the successor to NAT-PMP, has similar protocol concepts and packet formats, but adds support for **IPv6 port maps** and options/extensions. For more information, see:
**Port Mapping Protocols Overview and Comparison 2024 - About UPnP IGD & PCP/NAT-PMP**
https://github.com/Self-Hosting-Group/wiki/wiki/Port-Mapping-Protocols-Overview

Maintainer: @jow-

### CODE REVIEW

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 1]
```diff
@@ -107,97 +107,108 @@ return view.extend({
...
-			_('Value in KByte/s, informational only')).rmempty = true
+		o = s.taboption('general', form.Value, 'upload', _('Upload speed'),
+			_('Report maximum upload speed in kByte/s'))
+		o.rmempty = true
+		o.depends('enable_upnp', '1')
 
-		o = s.taboption('general', form.Value, 'port', _('Port'))
-		o.datatype = 'port'
-		o.default  = 5000
+		s.taboption('advanced', form.Flag, 'use_stun', _('Use %s', 'Use %s (%s = STUN)').format('<a href="https://en.wikipedia.org/wiki/STUN" target="_blank" rel="noreferrer"><abbr title="Session Traversal Utilities for NAT">STUN</abbr></a>'), _('To detect the public IPv4 address for unrestricted full-cone/one-to-one NATs'))
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:142]: does this setting exist in the openwrt package?

Comment (systemcrash) [reply applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:142]: OK good.

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 2]
```diff
@@ -73,18 +73,18 @@ return view.extend({
 		var m, s, o;
 
 		m = new form.Map('upnpd', [_('UPnP IGD & PCP/NAT-PMP Service')],
-			_('UPnP IGD & PCP/NAT-PMP allows clients on the local network to automatically configure port forwards on the router. Also called Universal Plug and Play.'));
+			_('The %s & %s/%s protocols allow clients on the local network to configure port maps/forwards on the router autonomously.', 'The %s (%s = UPnP IGD) & %s (%s = PCP)/%s (%s = NAT-PMP) protocols allow clients on the local network to configure port maps/forwards on the router autonomously.').format('<a href="https://en.wikipedia.org/wiki/Internet_Gateway_Device_Protocol" target="_blank" rel="noreferrer"><abbr title="UPnP Internet Gateway Device (Control Protocol)">UPnP IGD</abbr></a>', '<a href="https://en.wikipedia.org/wiki/Port_Control_Protocol" target="_blank" rel="noreferrer"><abbr title="Port Control Protocol">PCP</abbr></a>', '<a href="https://en.wikipedia.org/wiki/NAT_Port_Mapping_Protocol" target="_blank" rel="noreferrer"><abbr title="NAT Port Mapping Protocol">NAT-PMP</abbr></a>'));
```
[the remark this answers is not in the document]
Comment (systemcrash) [reply applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:76]: Exactly so.

[the remark this answers is not in the document]
Comment (systemcrash) [reply applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:76]: Ampersand is widely understood. But some languages eschew it entirely. Here it's acceptable.

**Comment by systemcrash** [comment]
rebase, squash please

**Comment by systemcrash** [comment]
keep it around

**Comment by systemcrash** [comment]
> Last but not least.
> 
>     1. I had renamed general settings to general setup, but now to service setup, with the corresponding advanced settings. Good choice?

In so doing we orphan a bunch of already good i18n, with no obvious improvement. Just seems synonymous. 

 
>     2. The UPnP IGD & PCP/NAT-PMP protocols allow clients on the local network to configure port maps/forwards on the router _autonomously_. Or: The UPnP IGD & PCP/NAT-PMP protocols allow clients on the local network to configure _autonomously_ port maps/forwards on the router. Autonomous is the best word, better than e.g. independent?

They're more or less synonyms. Either.

>     3. Would a help text in the UI with e.g. two example STUN servers be useful or in the wiki?

??

**Comment by systemcrash** [comment]
So, you've tested these changes in situ, correct? What happens when you edit an ACL, delete one of the port fields, and press save?

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 3]
```diff
@@ -107,97 +107,108 @@ return view.extend({
...
 		s.tab('advanced', _('Advanced Settings'));
 
-		o = s.taboption('general', form.Flag, 'enabled', _('Start service'));
+		o = s.taboption('setup', form.Flag, 'enabled', _('Start service'), _('Start autonomous port mapping service'));
 		o.rmempty  = false;
 
-		s.taboption('general', form.Flag, 'enable_upnp', _('Enable UPnP IGD protocol')).default = '1'
-		s.taboption('general', form.Flag, 'enable_natpmp', _('Enable PCP/NAT-PMP protocol')).default = '1'
+		s.taboption('setup', form.Flag, 'enable_upnp', _('Enable UPnP IGD protocol')).default = '1'
+		s.taboption('setup', form.Flag, 'enable_natpmp', _('Enable PCP/NAT-PMP protocols')).default = '1'
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:125]: terminate every expression with `;` (same for all remaining lines in this file)

**Comment by systemcrash** [comment]
> So, you've tested these changes in situ, correct? What happens when you edit an ACL, delete one of the port fields, and press save?

?

**Comment by systemcrash** [comment]
> > > So, you've tested these changes in situ, correct? What happens when you edit an ACL, delete one of the port fields, and press save?
> > 
> > 
> > ?
> 
> Yes! Please check the completed code.

Please test and answer my question :)

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 4]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+		o = s.option(form.Value, 'int_addr', _('Client Address'));
+		o.datatype = 'ip4addr';
+		o.placeholder = '0.0.0.0/0';
+		o.validate = function (section_id, value) {
+			return value !== "";
+		};
+
+		o = s.option(form.Value, 'int_ports', _('Client Port'));
+		o.datatype = 'portrange';
+		o.default = '1-65535';
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:212]:
```js
		o.placeholder = '1-65535';
```

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 5]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+		o.validate = function (section_id, value) {
+			return value !== "";
+		};
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:215]: drop

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 6]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+
+		o = s.option(form.Value, 'ext_ports', _('External Port'));
+		o.datatype = 'portrange';
+		o.default = '1-65535';
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:219]:
```js
		o.placeholder = '1-65535';
```

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 7]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+		o.validate = function (section_id, value) {
+			return value !== "";
+		};
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:222]: drop

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 8]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+		s.addremove = true;
+
+		s.option(form.Value, 'comment', _('Comment'));
+
+		o = s.option(form.Value, 'int_addr', _('Client Address'));
+		o.datatype = 'ip4addr';
+		o.placeholder = '0.0.0.0/0';
+		o.validate = function (section_id, value) {
+			return value !== "";
+		};
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:208]: drop

File: applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js [hunk 9]
```diff
@@ -107,99 +107,121 @@ return view.extend({
...
+		s = m.section(form.GridSection, 'perm_rule', _('Service Access Control List'),
+			_('ACL specify which client addresses and ports can be mapped, IPv6 always allowed.'));
+		s.sortable = true;
+		s.anonymous = true;
+		s.addremove = true;
+
+		s.option(form.Value, 'comment', _('Comment'));
+
+		o = s.option(form.Value, 'int_addr', _('Client Address'));
+		o.datatype = 'ip4addr';
```
Comment (systemcrash) [inline applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js:204]:
```js
		o.datatype = 'ipaddr';
```

if miniupnpd supports ipv6 also

**Comment by systemcrash** [comment]
OK. Done I think. Rebase, squash your commits, please.

**Comment by systemcrash** [comment]
No merge commits. Rebase on master.

**Comment by hnyman** [comment]
@systemcrash 
Due to this PR being merged, there is a weblate conflict.

```
Rebasing (1/1)
Auto-merging applications/luci-app-upnp/po/zh_Hant/upnp.po
CONFLICT (content): Merge conflict in applications/luci-app-upnp/po/zh_Hant/upnp.po
error: could not apply 428a43b365... Translated using Weblate (Chinese (Traditional Han script))
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 428a43b365... Translated using Weblate (Chinese (Traditional Han script))
 (1)
```

Stokito's later PR that you also merged, did not help, as the weblate was already in conflict.

**Comment by systemcrash** [comment]
I'll get that fixed (if you didn't already)

**Comment by hnyman** [comment]
I have been away from my PC, so I haven't fixed it

**Comment by systemcrash** [comment]
Fixed. Thanks for the heads-up @hnyman

**Pull request closed without merging**